docs/modules.md

```diff
@@ -347,6 +437,22 @@ The module's author should also update any example in the module's configuration
 use the new `modules` section in Synapse's configuration file (see [this section](#using-modules)
 for more info).
 
+#### Porting password auth provider modules
+
+There is no longer a `get_db_schema_files` callback provided, any changes to the database
+should now be made by the module using the module API class.
+
+To port a module that has a `check_password` method: 
+- Register `("m.login.password", ("password",): self.check_password` as an auth checker
+- Set `self.module_api` to point to the `ModuleApi` object given in `__init__`
```

Comment:
Why is this step necessary? It shouldn't matter how a module calls the module API instance it's given.